modules/default/compliments/compliments.js

```diff
@@ -38,11 +40,40 @@ Module.register("compliments", {
 			this.config.compliments = JSON.parse(response);
 			this.updateDom();
 		}
+		let minute_sync_delay = 1;
+		// loop thru all the configured when events
+		for (let m of Object.keys(this.config.compliments)) {
+			// if one contains a space, its a cron entry
+			if (m.includes(" ")) {
+				// we need to synch our interval cycle to the minute
+				minute_sync_delay = (60 - (moment().second())) * 1000;
+				break;
+			}
+		}
+		// Schedule update timer. sync to the minute start (if needed), so minute based events happen on the minute start
+		setTimeout(() => {
+			setInterval(() => {
+				this.updateDom(this.config.fadeSpeed);
+			}, this.config.updateInterval); },
+		minute_sync_delay);
+	},
+
+	/**
+	 * @param {string} cronExpression The cron expression. See https://croner.56k.guru/usage/pattern/
+	 * @param {Date} [timestamp] The timestamp to check. Defaults to the current time.
+	 * @returns {number} The number of seconds until the next cron run.
+	 */
+	getSecondsUntilNextCronRun (cronExpression, timestamp = new Date()) {
+		// Required for seconds precision
+		const adjustedTimestamp = new Date(timestamp.getTime() - 1000);
 
-		// Schedule update timer.
-		setInterval(() => {
-			this.updateDom(this.config.fadeSpeed);
-		}, this.config.updateInterval);
+		// https://www.npmjs.com/package/croner
+		/*eslint-disable-next-line*/
+		const cronJob = new Cron(cronExpression);
+		const nextRunTime = cronJob.nextRun(adjustedTimestamp);
+
+		let secondsDelta = (nextRunTime - adjustedTimestamp) / 1000;
```

Comment:
can be a const?

Reply:
i'm still old school.. can't be a const if its set my code..  const is a literal.  (imho)

Reply:
fixed